yorkie/src/androidTest/kotlin/dev/yorkie/core/DocumentTest.kt

```diff
@@ -621,4 +623,215 @@ class DocumentTest {
         }
     }
+
+    @Test
+    fun test_single_document_event_stream_can_emit_document_status_changed() {
+        runBlocking {
+            val client = createClient()
+            val documentKey = UUID.randomUUID().toString().toDocKey()
+            val document = Document(documentKey)
+
+            val documentStatusChangedList = mutableListOf<Event.DocumentStatusChanged>()
+            val job =
+                "document status changed events" to launch(start = CoroutineStart.UNDISPATCHED) {
+                    document.events.filterIsInstance<Event.DocumentStatusChanged>()
+                        .collect {
+                            documentStatusChangedList.add(it)
+                        }
+                }
+
+            client.activateAsync().await()
+            val actorID = client.requireClientId()
+
+            // 1. Can receive DocumentStatus.Attached event when attached
+            client.attachAsync(document, syncMode = Manual).await()
+            assertEquals(1, documentStatusChangedList.size)
+            assertEquals(DocumentStatus.Attached, documentStatusChangedList[0].documentStatus)
+            assertEquals(actorID, documentStatusChangedList[0].actorID)
+
+            // 2. Can receive DocumentStatus.Detached event when detached
+            client.detachAsync(document).await()
+            assertEquals(2, documentStatusChangedList.size)
+            assertEquals(DocumentStatus.Detached, documentStatusChangedList[1].documentStatus)
+            assertEquals(actorID, documentStatusChangedList[1].actorID)
+
+            // 3. Attach could fail if the document has been detached
+            client.attachAsync(document).await()
+            assertEquals(2, documentStatusChangedList.size)
+            assertEquals(DocumentStatus.Detached, documentStatusChangedList[1].documentStatus)
+            assertEquals(actorID, documentStatusChangedList[1].actorID)
+
+            // 4. Exception should be thrown when trying to remove a detached document
+            val exception = assertFailsWith(YorkieException::class) {
+                client.removeAsync(document).await()
+            }
+            assertEquals(YorkieException.Code.ErrDocumentNotAttached, exception.code)
+
+            client.deactivateAsync().await()
+            document.close()
+            client.close()
+            job.second.cancel()
+        }
+    }
+
+    @Test
+    fun test_multiple_document_event_stream_can_emit_document_status_changed() {
+        createTwoClientsAndDocuments { client1, client2, document1, document2, _ ->
+            val document1StatusChangedList = mutableListOf<Event.DocumentStatusChanged>()
+            val document2StatusChangedList = mutableListOf<Event.DocumentStatusChanged>()
+            val jobs = mapOf(
+                "document1 status changed events" to launch(start = CoroutineStart.UNDISPATCHED) {
+                    document1.events.filterIsInstance<Event.DocumentStatusChanged>()
```

Comment:
just a trivial question, but is there any reason to use a `Map` instead of a `List? It seems like you're not using the key.

Reply:
I didn't intend to use `Map` for anything else. Actually, as your comment, there was no need to use `Map`. 

Also, other functions I've written don't need `Map` either, but I think it makes sense for naming purposes, so I'll keep it as it is for now.